jwe.go

```diff
@@ -104,29 +105,70 @@ func (obj JSONWebEncryption) computeAuthData() []byte {
 	return output
 }
 
+func containsKeyAlgorithm(needle KeyAlgorithm, haystack []KeyAlgorithm) bool {
+	for _, algorithm := range haystack {
+		if algorithm == needle {
+			return true
+		}
+	}
+	return false
+}
+
+func containsContentEncryption(needle ContentEncryption, haystack []ContentEncryption) bool {
+	for _, algorithm := range haystack {
+		if algorithm == needle {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
The KeyAlgorithm and ContentEncryption types are both defined to be equivalent to `string`, which is [comparable](https://pkg.go.dev/builtin#comparable), so you [should be able to](https://go.dev/play/p/TdbUZtYvTS8) use [slices.Contains](https://pkg.go.dev/slices#Contains) instead of these.

If we're worried about backwards compatibility with go1.17 and below... well we're already doing an API breaking change here and creating v4. I think it's fine for v4 to be go1.18+ only.

Even if you do want to keep these one-off methods, I think it is slightly confusing that their argument order is the opposite that of the stdlib's.

Reply:
Fixed the order! I do want to keep back compat for now.